pkg/pusher/pusher.go

```diff
@@ -163,6 +167,13 @@ LOOP:
 						return
 					}
 				}
+
+				_, err = crypto.Recover(receipt.Signature, receipt.Address.Bytes())
```

Comment:
at least comment here that this is in no way aaa validation on any sort. ECrecover returns without an  error to whatever text you give, so the public key needs  checking I guess  this  can come in the subsequent PR but just comment temporary  or something to  indicate this

Reply:
Once we have the ability to check that the storer node is in the right neighborhood (ie depth oracle) with the public key, we will add it here. Thanks for the heads up.

Reply:
I think that would be a nice to have here, if we could actually use the `Recover` call to extract the node overlay address. There are helper functions in the crypto package to create a `NewOverlayAddress`. Then if we could somehow also log this as part of the information logged in `L146`, like: `pusher: pushed chunk %s to node %s`, logging the chunk address and the overlay address that signed the reciept. I think this could also help us increase our observability while debugging problems on clusters.